drivers/serial/uart_eusart.c

```diff
@@ -0,0 +1,516 @@
+	if (err < 0) {
+		return err;
+	}
+
+	/* Init EUSART */
+	eusartInit.baudrate = uart_cfg->baudrate;
+	eusartInit.parity = uart_eusart_cfg2ll_parity(uart_cfg->parity);
+	eusartInit.stopbits = uart_eusart_cfg2ll_stopbits(uart_cfg->stop_bits);
+	eusartInit.databits = uart_eusart_cfg2ll_databits(uart_cfg->data_bits, uart_cfg->parity);
+	advancedSettings.hwFlowControl = uart_eusart_cfg2ll_hwctrl(uart_cfg->flow_ctrl);
+	eusartInit.advancedSettings = &advancedSettings;
+
+	EUSART_UartInitHf(config->eusart, &eusartInit);
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	config->irq_config_func(dev);
+#endif
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int uart_eusart_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	__maybe_unused const struct uart_eusart_config *config = dev->config;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+#ifdef EUSART_STATUS_TXIDLE
+		/* Wait for TX FIFO to flush before suspending */
+		while (!(EUSART_StatusGet(config->eusart) & EUSART_STATUS_TXIDLE)) {
+		}
+#endif
+		break;
+
+	case PM_DEVICE_ACTION_RESUME:
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+#endif
+
+static DEVICE_API(uart, uart_eusart_driver_api) = {
+	.poll_in = uart_eusart_poll_in,
+	.poll_out = uart_eusart_poll_out,
+	.err_check = uart_eusart_err_check,
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	.fifo_fill = uart_eusart_fifo_fill,
+	.fifo_read = uart_eusart_fifo_read,
+	.irq_tx_enable = uart_eusart_irq_tx_enable,
+	.irq_tx_disable = uart_eusart_irq_tx_disable,
+	.irq_tx_complete = uart_eusart_irq_tx_complete,
+	.irq_tx_ready = uart_eusart_irq_tx_ready,
+	.irq_rx_enable = uart_eusart_irq_rx_enable,
+	.irq_rx_disable = uart_eusart_irq_rx_disable,
+	.irq_rx_ready = uart_eusart_irq_rx_ready,
+	.irq_err_enable = uart_eusart_irq_err_enable,
+	.irq_err_disable = uart_eusart_irq_err_disable,
+	.irq_is_pending = uart_eusart_irq_is_pending,
+	.irq_update = uart_eusart_irq_update,
+	.irq_callback_set = uart_eusart_irq_callback_set,
+#endif
+};
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+#define UART_EUSART_IRQ_HANDLER_DECL(idx)				       \
+	static void uart_eusart_config_func_##idx(const struct device *dev)
+#define UART_EUSART_IRQ_HANDLER_FUNC(idx)				       \
+	.irq_config_func = uart_eusart_config_func_##idx,
+#define UART_EUSART_IRQ_HANDLER(idx)					       \
+	static void uart_eusart_config_func_##idx(const struct device *dev)   \
+	{								       \
+		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(idx, rx, irq),		       \
+			    DT_INST_IRQ_BY_NAME(idx, rx, priority),	       \
+			    uart_eusart_isr, DEVICE_DT_INST_GET(idx), 0);     \
+		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(idx, tx, irq),		       \
+			    DT_INST_IRQ_BY_NAME(idx, tx, priority),	       \
+			    uart_eusart_isr, DEVICE_DT_INST_GET(idx), 0);     \
+									       \
+		irq_enable(DT_INST_IRQ_BY_NAME(idx, rx, irq));		       \
+		irq_enable(DT_INST_IRQ_BY_NAME(idx, tx, irq));		       \
+	}
+#else
+#define UART_EUSART_IRQ_HANDLER_DECL(idx)
```

Comment:
`UART_EUSART_IRQ_HANDLER_DECL()` is orphan.

Reply:
removed it